test/e2e/create_test.go

```diff
@@ -262,16 +339,49 @@ func (tc *testContext) waitForWindowsNodes(nodeCount int32, waitForAnnotations,
 					node.Annotations[nodeconfig.PubKeyHashAnnotation], pubKeyAnnotation)
 				return false, nil
 			}
+
+		}
+		// Now verify that we have found all the nodes being waited for
+		if len(nodes) != int(nodeCount) {
+			log.Printf("waiting for %d/%d Windows nodes", len(nodes), nodeCount)
+			return false, nil
 		}
 
+		// Initialize/update nodes to avoid staleness
+		if isBYOH {
+			gc.byohNodes = nodes
+		} else {
+			gc.machineNodes = nodes
+		}
 		return true, nil
 	})
 
-	// Initialize/update nodes to avoid staleness
-	gc.nodes = nodes.Items
 	// Log the time elapsed while waiting for creation of the nodes
 	endTime := time.Now()
-	log.Printf("%v time is required to configure %v nodes", endTime.Sub(startTime), gc.numberOfNodes)
+	log.Printf("%v time is required to configure %v nodes", endTime.Sub(startTime), nodeCount)
 
 	return err
 }
+
+// listFullyConfiguredWindowsNodes returns a slice of nodes. If isBYOH is set to true, the nodes returned will be
+// BYOH nodes, else they will be nodes configured by the Machine controller.
+// A node is considered fully configured once it has the WMCO version annotation applied to it.
+func (tc *testContext) listFullyConfiguredWindowsNodes(isBYOH bool) ([]v1.Node, error) {
+	nodes, err := tc.client.K8s.CoreV1().Nodes().List(context.TODO(),
+		metav1.ListOptions{LabelSelector: v1.LabelOSStable + "=windows"})
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list nodes")
+	}
+	var windowsNodes []v1.Node
+	for _, node := range nodes.Items {
+		// filter out nodes that haven't been fully configured
+		if _, present := node.Labels[nodeconfig.VersionAnnotation]; !present {
+			continue
+		}
+		if (isBYOH && node.Annotations[controllers.BYOHAnnotation] == "true") ||
```

Comment:
a liitle confused as to why is this check required for appending, aren't we already filtering on the label?

Reply:
This check is figuring out if the node is a BYOH or Machine node. The previous checks were to ensure that 1) the node is Windows, 2) the node is fully configured. The BYOH/Machine check cannot be made until the node is fully configured, as the BYOH annotation is applied at the same time as the version annotation